code/modules/projectiles/guns/specialist.dm

```diff
@@ -715,121 +694,47 @@
 	icon_state = "m81"
 	item_state = "m81"
 	max_shells = 1 //codex
-	caliber = CALIBER_40MM //codex
-	load_method = SINGLE_CASING //codex
-	w_class = WEIGHT_CLASS_BULKY
 	flags_equip_slot = ITEM_SLOT_BACK|ITEM_SLOT_BELT
-	throw_speed = 2
-	throw_range = 10
-	force = 5.0
 	wield_delay = 0.2 SECONDS
-	fire_sound = 'sound/weapons/guns/fire/grenadelauncher.ogg'
-	fire_rattle = 'sound/weapons/guns/fire/grenadelauncher.ogg'
-	cocked_sound = 'sound/weapons/guns/interact/m92_cocked.ogg'
 	aim_slowdown = 1
-	gun_skill_category = GUN_SKILL_FIREARMS
 	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_AMMO_COUNTER
 	attachable_allowed = list()
-	general_codex_key = "explosive weapons"
 	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 18,"rail_x" = 14, "rail_y" = 22, "under_x" = 19, "under_y" = 14, "stock_x" = 19, "stock_y" = 14)
 	fire_delay = 1.05 SECONDS
-	/// The loaded grenade
-	var/obj/item/explosive/grenade/grenade
-	/// What type of grenade can be loaded
+	max_grenades = 1
 	var/grenade_type_allowed = /obj/item/explosive/grenade
 
-/obj/item/weapon/gun/launcher/m81/Initialize(mapload, spawn_empty)
+/obj/item/weapon/gun/launcher/m81/Initialize()
 	. = ..()
-	if(!spawn_empty)
-		grenade = new grenade_type_allowed()
+	update_icon() //we adapt in case it starts empty or full
 
 /obj/item/weapon/gun/launcher/m81/update_icon()
-	icon_state = grenade ? base_gun_icon : base_gun_icon + "_e"
-
+	icon_state = "[base_gun_icon][length(grenades) ? "" : "_e"]"
 
 /obj/item/weapon/gun/launcher/m81/examine_ammo_count(mob/user)
-	if(!grenade || (get_dist(user, src) > 2 && user != loc))
+	if(!length(grenades) || (get_dist(user, src) > 2 && user != loc))
 		return
-	to_chat(user, span_notice("It is loaded with [grenade]."))
-
+	to_chat(user, span_notice("It is loaded with [grenades[1]]."))
 
 /obj/item/weapon/gun/launcher/m81/attackby(obj/item/I, mob/user, params)
-	if(!istype(I, /obj/item/explosive/grenade))
-		return ..()
-
-	if(!istype(I, grenade_type_allowed))
-		to_chat(user, span_warning("[src] can't use [I]!"))
-		return
-
-	if(grenade)
-		to_chat(user, span_warning("[src] cannot hold more grenades!"))
-		return
-
-	if(!user.transferItemToLoc(I, src))
-		return
-
-	grenade = I
-	to_chat(user, span_notice("You put [I] in [src]."))
+	. = ..()
 	update_icon()
 
-/obj/item/weapon/gun/launcher/m81/afterattack(atom/target, mob/user, flag)
-	if(!able_to_fire(user))
-		return
-	if(gun_on_cooldown(user))
-		return
-	if(get_dist(target,user) <= 2)
-		to_chat(user, span_warning("[src] beeps a warning noise. You are too close!"))
-		return
-	if(!grenade)
-		to_chat(user, span_warning("[src] is empty."))
-		return
-	fire_grenade(target, user)
-	playsound(user.loc, cocked_sound, 25, 1)
+/obj/item/weapon/gun/launcher/m81/unload(mob/user)
+	. = ..()
+	update_icon()
```

Comment:
move update_icon() up to the end of the parent proc, and delete this here. Its not needed and the parent unload() should call update_icon(), not this one